src/Nethermind/Nethermind.UPnP.Plugin/UPnPPlugin.cs

```diff
@@ -48,7 +47,8 @@ private async Task RunRefreshLoop()
                 if (_logger.IsWarn) _logger.Error("Unable to setup UPnP mapping.", exception);
             }
 
-            if (await _cancellationTokenSource.Token.WaitHandle.WaitOneAsync(ExpirationRate, CancellationToken.None))
+            await _timer.WaitForNextTickAsync(_cancellationTokenSource.Token);
+            if (_cancellationTokenSource.Token.IsCancellationRequested)
```

Comment:
You can put that directly in a while check instead of having separate `if` with `break`
`while (await timer.WaitForNextTickAsync(cancellationToken))`